src/Numerics/ODESolvers/DifferentialEquations.jl

```diff
@@ -0,0 +1,115 @@
+via a DifferentialEquations.jl DEAlgorithm, which includes support
+for OrdinaryDiffEq.jl, Sundials.jl, and more.
+"""
+mutable struct DiffEqJLSolver{I} <: AbstractDiffEqJLSolver
+    integ::I
+
+    function DiffEqJLSolver(rhs!,alg,Q,args...;
+                            t0 = 0, p = nothing,
+                            kwargs...)
+        prob = DiffEqBase.ODEProblem(
+                          (du,u,p,t)->rhs!(du,u,p,t; increment=false),
+                           Q, (float(t0),typemax(typeof(float(t0)))),
+                           p)
+        integ = DiffEqBase.init(prob,alg,args...;kwargs...)
+        new{typeof(integ)}(integ)
+    end
+end
+
+"""
+    DiffEqJLSolver(f, RKA, RKB, RKC, Q; dt, t0 = 0)
+
+This is a time stepping object for explicitly time stepping the differential
+equation given by the right-hand-side function `f` with the state `Q`, i.e.,
+
+```math
+  \\dot{Q} = f_I(Q, t) + f_E(Q, t)
+```
+
+via a DifferentialEquations.jl DEAlgorithm, which includes support
+for OrdinaryDiffEq.jl, Sundials.jl, and more.
+"""
+mutable struct DiffEqJLIMEXSolver{I} <: AbstractDiffEqJLSolver
+    integ::I
+
+    function DiffEqJLIMEXSolver(rhs!,rhs_implicit!,alg,Q,args...;
+                            t0 = 0,
+                            kwargs...)
+        prob = SplitODEProblem(
+                          (du,u,p,t)->rhs_implicit!(du,u,p,t; increment=false),
+                          (du,u,p,t)->rhs!(du,u,p,t; increment=false),
+                           Q, (float(t0),typemax(typeof(float(t0)))),
+                           p)
+        integ = DiffEqBase.init(prob,alg,args...;kwargs...)
+        new{typeof(integ)}(integ)
+    end
+end
+
+gettime(solver::AbstractDiffEqJLSolver) = solver.integ.t
+getdt(solver::AbstractDiffEqJLSolver) = solver.integ.dt
+updatedt!(solver::AbstractDiffEqJLSolver, dt) = DiffEqBase.set_proposed_dt!(solver.integ,dt)
+updatetime!(solver::AbstractDiffEqJLSolver, t) = DiffEqBase.set_t!(solver.integ,t)
+isadjustable(solver::AbstractDiffEqJLSolver) = true # Is this isadaptive? Or something different?
+
+"""
+    ODESolvers.general_dostep!(Q, solver::AbstractODESolver, p,
+                               timeend::Real, adjustfinalstep::Bool)
+
+Use the solver to step `Q` forward in time from the current time, to the time
+`timeend`. If `adjustfinalstep == true` then `dt` is adjusted so that the step
+does not take the solution beyond the `timeend`.
+"""
+function general_dostep!(
+    Q,
+    solver::AbstractDiffEqJLSolver,
+    p,
+    timeend::Real;
+    adjustfinalstep::Bool,
+)
+    integ = solver.integ
+
+    if DiffEqBase.DataStructures.top(integ.opts.tstops) !== timeend
+        DiffEqBase.add_tstop!(integ,timeend)
+    end
+    dostep!(Q, solver, p, time)
+    solver.integ.t
+end
+
+function dostep!(
+    Q,
+    solver::AbstractDiffEqJLSolver,
+    p,
+    time,
+    slow_δ = nothing,
+    slow_rv_dQ = nothing,
+    in_slow_scaling = nothing,
+)
+
+    integ = solver.integ
+    integ.p = p # Can this change?

```

Comment:
Sorry for the confusion! Sadly, `p` is never really documented or used by most ODESolvers. It's only ever used in special circumstances where we are nesting RK methods within, say, a multirate or multirate infinitesimal-step (only the GARK solvers right now) time-stepping method. It's more or less a context manager for passing information through nested ODESolvers.

So in answer to your question on whether it can change: it may certainly change if this particular ODESolver becomes part of a multi-rate method. However, I am not sure what this would look like using the differential evolution algorithm (I assume this is what you call the DE algorithm correct?).

My experience with multirate is purely within the Runge-Kutta framework.

Reply:
Okay. Hmm, if it doesn't have the parameters information then we won't be able to do adjoint sensitivities. I'll want to talk to someone about that, but that can be the next thing to do.